CDCgov/prime-reportstream
# 14115: further convert cleanup

This PR adds some additional logging to the convert step and adds the logs from the convert step into the submission endpoint

Test Steps:
1. Send this message
```sh
curl --location 'http://localhost:7071/api/reports' \
--header 'client: ignore.ignore-mars-otc' \
--header 'Content-Type: application/hl7-v2' \
--data 'MSH|^~\&|MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|CAREEVOLUTION^00Z0000024^CLIA|AIMS.INTEGRATION.PRD^2.16.840.1.114222.4.3.15.1^ISO|AIMS.PLATFORM^2.16.840.1.114222.4.1.217446^ISO|20240403205305+0000||ORU^R01^ORU_R01|20240403205305_dba7572cc6334f1ea0744c5f235c823e|P|2.5.1|||NE|NE|||||PHLabReport-NoAck^ELR251R1_Rcvr_Prof^2.16.840.1.113883.9.11^ISO
SFT|CAREEVOLUTION|2022|MMTC.PROD|16498||20240402
PID|1||8be6fa3710374dcebe0174e0fd5a1a7c^^^MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO^PI||^^^^^^~^^^^^^||||||^^^^02139^USA||^^^^^111^1111111
ORC|RE||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|||||||||^^||^^^^^^|||||||SA.OTCSelfReport|^^^^02139^^^^|^^^^^^
OBR|1||^MMTC.PROD^2.16.840.1.113883.3.8589.4.2.106.1^ISO|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71|||20240403120000|||||||||^^|^^^^^^|||||20240403120000|||F
OBX|1|CWE|94558-4^SARS-CoV-2 (COVID-19) Ag [Presence] in Respiratory specimen by Rapid immunoassay^LN^^^^2.71||260373001^Detected^SCT^^^^20200901||||||F||||00Z0000042||BinaxNOW COVID-19 Antigen Self Test_Abbott Diagnostics Scarborough, Inc._EUA^^99ELR^^^^Vunknown||20240403120000||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042|
NTE|1|L|Note
OBX|2|NM|35659-2^Age at specimen collection^LN^^^^2.71||24|a^year^UCUM^^^^2.1|||||F||||00Z0000042||||||||SA.OTCSelfReport^^^^^&2.16.840.1.113883.3.8589.4.1.152&ISO^XX^^^00Z0000042||||||QST
SPM|1|^dba7572cc6334f1ea0744c5f235c823e&MMTC.PROD&2.16.840.1.113883.3.8589.4.2.106.1&ISO||697989009^Anterior nares swab^SCT^^^^20200901|||||||||||||20240403120000-0400|20240403120000-0400'
```
2. Watch the terminal and verify that the logs from the convert step have the report id and topic (look for `"TOPIC":"\"mars-otc-elr\""`)
3. Grab the outputted idea and hit the submission endpoint
```sh
curl --location 'http://localhost:7071/api/waters/report/<REPORTID>/history' \
--header 'authentication-type: okta' \
--header 'Authorization: Bearer dummy'
```
4. Verify that the validation error is in the response `"Item 1 in the report was not valid. Reason: HL7 was not valid at OBX[1]-19[1].1"`

## Changes
- Updates the logging in the convert step
- Properly generates a report if no bundles will be routed
- Updates the submission endpoint to return logs from the convert step

## Checklist

### Testing

### Process

## Linked Issues
- #14115 

## To Be Done
https://github.com/CDCgov/prime-reportstream/issues/14350
https://github.com/CDCgov/prime-reportstream/issues/14349


## Specific Security-related subjects a reviewer should pay specific attention to
